### DISCUSSION TITLE
Migrating to SQL alchemy 2.0 weird error log

### DISCUSSION THREAD

**Discussion by Randyklose** [sqlalchemy/sqlalchemy#11505] [Usage Questions]
My team is migrating to version 2.0 and we enabled the "RemovedIn20Warning" in prod. We are seeing the following error log but not sure what the  "<string>:1: " means. any ideas? We are able to see the file and line for the rest of errors but not for this one.

`<string>:1: RemovedIn20Warning: Deprecated API features detected! These feature(s) are not compatible with SQLAlchemy 2.0. To prevent incompatible upgrades prior to updating applications, ensure requirements files are pinned to "sqlalchemy<2.0". Set environment variable SQLALCHEMY_WARN_20=1 to show all deprecation warnings.  Set environment variable SQLALCHEMY_SILENCE_UBER_WARNING=1 to silence this message. (Background on SQLAlchemy 2.0 at: https://sqlalche.me/e/b8d9)`

We have been looking into the endpoints with the similar timestamps as the error log and tried adding unit tests to see if we can identify what is wrong but no warnings are shown when we run pytest with the flag on. 

There is a pattern in all the endpoints for which this warning appears afterwards which is we conditionally add something to a session. Then we do "flush" and then serialize and at the end we "commit" Not sure if that is part of the issue or just coincidence.

**Comment by zzzeek** [comment]
that ":1:" is a line number for Python code that is inside of a string, which we would assume was generated using eval / exec.    So it's code that was rendered into a string somewhere.    It's very possible a warning emitted from within SQLAlchemy in a place that we didn't tend to line numbers correctly, but could also be from the user-facing code.   When I test the warning locally, line numbers are emitted correctly.

however, this is a special warning we call the "uber warning" which just indicates that deprecated API features have been detected, and that the SQLALCHEMY_WARN_20 parameter is **not** turned on.  If you have SQLALCHEMY_WARN_20 turned on, then you would not see this warning, you would see the individual warnings for your code.    So where this is running, SQLALCHEMY_WARN_20 is not actually being detected.

To know exactly where this warning comes from, set the warnings filter to "error" and it will just throw an error, you'll get a complete stack trace.
